File: week1/GamGomYang/BOJ1978.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class BOJ1978 {

    public static boolean primeSolution(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i <= num; i++) { 
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static int primeCounter(List<Integer> numbers) {
        int counter = 0;
        for (int number : numbers) {
            if (primeSolution(number)) {
                counter++;
            }
        }
        return counter;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        List<Integer> numbers = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            numbers.add(scanner.nextInt());
        }

        int counter = primeCounter(numbers);

        System.out.println(counter);
    }
}

File: week1/GamGomYang/BOJ1978.py
from typing import List

def prime_solution(num: int) -> bool:
    if num <= 1:
        return False
    for i in range(2, num):
        if num % i == 0:
            return False
    return True

def prime_counter(numbers: List[int]) -> int:
    counter = 0
    for number in numbers:
        if prime_solution(number):
            counter += 1
    return counter


n = int(input())
numbers = list(map(int, input().split()))


counter = prime_counter(numbers)


print(counter)
